src/main/java/com/facebook/openwifirrm/modules/ModelerUtils.java

```diff
@@ -381,4 +386,150 @@ public static Map<String, Map<String, WifiScanEntry>> getAggregatedWifiScans(
 		}
 		return aggregatedWifiScans;
 	}
+
+	/** This method converts the input State info to an AggregatedState
+	 * and adds it to the bssidToAggregatedStates map. If the bssid&station
+	 * of the input State does not exist in the map, create a new
+	 * AggregatedState list. If the bssid&station of the input State exists,
+	 * then convert State to AggregatedState and check if there exits an
+	 * AggregatedState of the same radio. If there does, append the value
+	 * of aggregation field to the existing AggregatedState, if not, create
+	 * a new AggregatedState and add it to the list.
+	 *
+	 * @param bssidToAggregatedStates map from bssid&station to a list of AggregatedState.
+	 * @param state the state that is to be added.
+	 */
+	static void addStateToAggregation(
+		Map<String, List<AggregatedState>> bssidToAggregatedStates,
+		State state
+	) {
+		for (Interface stateInterface : state.interfaces) {
+			if (stateInterface.ssids == null) {
+				continue;
+			}
+			for (SSID ssid : stateInterface.ssids) {
```

Comment:
Maybe more just personal preference: I think it would make sense to take this second for-loop and move it to its own method (which takes in `bssidToAggregatedStates` and `stateInterface`)? Mainly just to try and make this method less "nested"